Framework/Beamline/src/ComponentInfo.cpp

```diff
@@ -111,16 +111,45 @@ Eigen::Quaterniond ComponentInfo::rotation(const size_t componentIndex) const {
   return (*m_rotations)[rangesIndex];
 }
 
+/**
+ * Extract the position of a component relative to it's parent
+ *
+ * The parent rotatation is unwound prior to establishing the offset. This means
+ *that
+ * recorded relative positions are independent of changes in rotation.
+ *
+ * BEWARE that this method does not account for scaling factors as found in
+ *RectangularDetectors
+ * see Instrument::makeLegacyParmeterMap for correct handling of those cases.
+ *
+ * BEWARE of peformance on repeated calls to this method as the transformation
+ *has to
+ * be established every time, independent of whether the parent remains the
+ *same.
+ *
+ * @param componentIndex
+ * @return
+ */
 Eigen::Vector3d
 ComponentInfo::relativePosition(const size_t componentIndex) const {
   size_t parentIndex = parent(componentIndex);
   if (parentIndex == componentIndex) {
     return position(componentIndex);
   } else {
-    return position(componentIndex) - position(parentIndex);
+    const auto parentPos = position(parentIndex);
+    Eigen::Affine3d transformation;
+    transformation =
+        rotation(parentIndex).conjugate(); // Inverse parent rotation
+    transformation.translate(-parentPos);
+    return transformation * position(componentIndex);
```

Comment:
I have found that using `Affine3d` is slower than doing an offset + rotation. If you use this in performance critical code consider simply writing `invrot *(position - parentPos)`.